pkg/scheduler/internal/cache/node_tree.go

```diff
@@ -29,31 +30,16 @@ import (
 // NodeTree is NOT thread-safe, any concurrent updates/reads from it must be synchronized by the caller.
 // It is used only by schedulerCache, and should stay as such.
 type nodeTree struct {
-	tree      map[string]*nodeArray // a map from zone (region-zone) to an array of nodes in the zone.
-	zones     []string              // a list of all the zones in the tree (keys)
-	zoneIndex int
-	numNodes  int
+	tree     map[string]*nodeArray // a map from zone (region-zone) to an array of nodes in the zone.
+	zones    []string              // a list of all the zones in the tree (keys)
+	numNodes int
 }
 
 // nodeArray is a struct that has nodes that are in a zone.
 // We use a slice (as opposed to a set/map) to store the nodes because iterating over the nodes is
 // a lot more frequent than searching them by name.
 type nodeArray struct {
-	nodes     []string
-	lastIndex int
-}
-
-func (na *nodeArray) next() (nodeName string, exhausted bool) {
-	if len(na.nodes) == 0 {
-		klog.Error("The nodeArray is empty. It should have been deleted from NodeTree.")
-		return "", false
-	}
-	if na.lastIndex >= len(na.nodes) {
-		return "", true
-	}
-	nodeName = na.nodes[na.lastIndex]
-	na.lastIndex++
-	return nodeName, false
+	nodes []string
```

Comment:
I'd like to define `nodeTree.tree` as `map[string][]string`, so `nodeArray` can be eliminated.

Reply:
done, thank you